### ISSUE TITLE
xud tries to connect to 0.0.0.0 addresses

### ISSUE THREAD

**Issue by kilrau** [ExchangeUnion/xud#1091]
### Background

xud tries to connect to 0.0.0.0 addresses 

### Your environment

* version of `xud`: 1.0.0-alpha.11
* which operating system (`uname -a` on *Nix): ubuntu 18.04
* version of `btcd`, `bitcoind`, `ltcd`, `litecoind`, `geth`, `parity`, `raiden`, `lnd` or other chain backend: xud simnet setup

### Steps to reproduce

Run xud simnet, check xud log

### Expected behaviour

xud tries to connect to real IP addresses only

### Actual behaviour

xud tries to connect to 0.0.0.0:8885, which is not a valid address and includes itself:
```
11/07/2019 10:04:54.113 [P2P] debug: Connected pre-handshake to peer 0.0.0.0:8885
11/07/2019 10:04:54.117 [P2P] debug: received disconnecting packet from 0.0.0.0:8885:{"reason":11}
11/07/2019 10:04:54.118 [P2P] info: Socket closed prior to handshake (0.0.0.0:8885)
```
Oldie issue back again, see https://github.com/ExchangeUnion/xud/issues/177

**Comment by sangaman** [comment]
Do you have advertised addresses? If so, we attempt a connection to ourselves for each advertised address once only when starting up. Just want to make sure that's not the case here, did those messages come shortly after a `Verifying reachability of advertised address:` line in the logs?

**Comment by kilrau** [comment]
Correct, that's exactly the case, see below @sangaman . Not a big issue then, but maybe we should just exclude trying to connect to "special addresses" like 127.0.01 and 0.0.0.0.

```
xud_1        | 23/07/2019 16:21:42.422 [P2P] info: Connecting to testnet XU network
xud_1        | 23/07/2019 16:21:42.426 [P2P] info: p2p server listening on 0.0.0.0:8885
xud_1        | 23/07/2019 16:21:42.428 [P2P] debug: Verifying reachability of advertised address: 0.0.0.0:8885
xud_1        | 23/07/2019 16:21:42.434 [HTTP] info: http server listening on 0.0.0.0:8887
xud_1        | 23/07/2019 16:21:42.441 [P2P] debug: Connected pre-handshake to peer 127.0.0.1:44446
xud_1        | 23/07/2019 16:21:42.535 [P2P] debug: Peer (030c74ebef2068dfa48c8cae1a33b670528cf10d636d11f8f9ee61e89f02aba93e): closing socket. reason: ConnectedToSelf
xud_1        | 23/07/2019 16:21:42.537 [P2P] warn: could not open connection to peer (030c74ebef2068dfa48c8cae1a33b670528cf10d636d11f8f9ee61e89f02aba93e): cannot attempt connection to self
xud_1        | 23/07/2019 16:21:42.538 [P2P] info: Peer 030c74ebef2068dfa48c8cae1a33b670528cf10d636d11f8f9ee61e89f02aba93e socket closed
xud_1        | 23/07/2019 16:21:42.542 [P2P] verbose: Verified reachability of advertised address: 0.0.0.0:8885
xud_1        | 23/07/2019 16:21:42.544 [RPC] info: gRPC server listening on 0.0.0.0:8886
xud_1        | 23/07/2019 16:21:42.547 [P2P] info: Connecting to known / previously connected peers
xud_1        | 23/07/2019 16:21:42.624 [P2P] debug: Connected pre-handshake to peer xud1.testnet.exchangeunion.com:8885
xud_1        | 23/07/2019 16:21:42.667 [P2P] debug: received disconnecting packet from xud1.testnet.exchangeunion.com:8885:{"reason":11}
xud_1        | 23/07/2019 16:21:42.667 [P2P] info: Socket closed prior to handshake (xud1.testnet.exchangeunion.com:8885)
xud_1        | 23/07/2019 16:21:42.668 [P2P] warn: could not open connection to peer (xud1.testnet.exchangeunion.com:8885): Peer (xud1.testnet.exchangeunion.com:8885) disconnected from us due to AuthFailureInvalidTarget
xud_1        | 23/07/2019 16:21:42.670 [P2P] debug: Connected pre-handshake to peer 127.0.0.1:44450
xud_1        | 23/07/2019 16:21:42.671 [P2P] debug: Connected pre-handshake to peer 0.0.0.0:8885
xud_1        | 23/07/2019 16:21:42.684 [P2P] debug: Peer (127.0.0.1:44450): reputation event: InvalidAuth
xud_1        | 23/07/2019 16:21:42.684 [P2P] debug: Peer (127.0.0.1:44450): closing socket. reason: AuthFailureInvalidTarget
xud_1        | 23/07/2019 16:21:42.685 [P2P] warn: could not open connection to peer (127.0.0.1:44450): could not authenticate peer (030c74ebef2068dfa48c8cae1a33b670528cf10d636d11f8f9ee61e89f02aba93e): invalid target (02d8ac24fe2952d0d4c9eaa188a1d51ae4d80879fa311a3729a23a15b24f121d6c)
xud_1        | 23/07/2019 16:21:42.686 [P2P] info: Socket closed prior to handshake (127.0.0.1:44450)
xud_1        | 23/07/2019 16:21:42.687 [P2P] debug: received disconnecting packet from 0.0.0.0:8885:{"reason":11}
xud_1        | 23/07/2019 16:21:42.688 [P2P] info: Socket closed prior to handshake (0.0.0.0:8885)
xud_1        | 23/07/2019 16:21:42.688 [P2P] warn: could not open connection to peer (0.0.0.0:8885): Peer (0.0.0.0:8885) disconnected from us due to AuthFailureInvalidTarget
xud_1        | 23/07/2019 16:21:42.719 [P2P] debug: Connected pre-handshake to peer xud1.testnet.exchangeunion.com:8885
xud_1        | 23/07/2019 16:21:42.765 [P2P] debug: received disconnecting packet from xud1.testnet.exchangeunion.com:8885:{"reason":11}
xud_1        | 23/07/2019 16:21:42.765 [P2P] info: Socket closed prior to handshake (xud1.testnet.exchangeunion.com:8885)
xud_1        | 23/07/2019 16:21:42.766 [P2P] warn: could not open connection to peer (xud1.testnet.exchangeunion.com:8885): Peer (xud1.testnet.exchangeunion.com:8885) disconnected from us due to AuthFailureInvalidTarget
xud_1        | 23/07/2019 16:21:42.766 [P2P] info: Completed start-up connections to known peers
xud_1        | 23/07/2019 16:22:13.769 [RPC] debug: received call /xudrpc.Xud/GetInfo
```
